.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,15 @@
+name: Unit Tests
+on:
+  push:
+    branches: [ main, master ]
+  pull_request:
+    branches: [ main, master ]
+jobs:
+  test:
+    runs-on: ubuntu-latest
```

Comment:
What version of node js does this run on? I'd expect us to test on at least the same versions that we support,

Probably that's at least 18, 20, 22

Reply:
No, this should be a matrix, and should test node 20, 18, and 22

Reply:
@azmy60 see how we do it in queryleaf - https://github.com/beekeeper-studio/queryleaf/blob/main/.github/workflows/test.yml

Reply:
Done!